frontend/packages/dev-console/src/components/topology/componentUtils.ts

```diff
@@ -170,7 +191,9 @@ const graphWorkloadDropTargetSpec: DropTargetSpec<
   canDrop: (item, monitor, props) => {
     return (
       monitor.isOver({ shallow: monitor.getItemType() === CREATE_CONNECTOR_DROP_TYPE }) &&
-      ((monitor.getOperation() === REGROUP_OPERATION && item.getParent() !== props.element) ||
+      ((monitor.getOperation() === REGROUP_OPERATION &&
+        item.hasParent() &&
```

Comment:
Could you add a comment to `item.hasParent()` check regarding the need for a fix for the async actions.